databricks-sdk-java/src/main/java/com/databricks/sdk/core/commons/CommonsHttpClient.java

```diff
@@ -57,6 +76,99 @@ private CloseableHttpClient makeClosableHttpClient() {
         .build();
   }
 
+  private CloseableHttpClient makeClosableHttpClient(ProxyConfig proxyConfig) {
+    HttpClientBuilder builder =
+        HttpClientBuilder.create()
+            .setConnectionManager(connectionManager)
+            .setDefaultRequestConfig(makeRequestConfig());
+    setupProxy(proxyConfig, builder);
+    return builder.build();
+  }
+
+  public static void setupProxy(ProxyConfig config, HttpClientBuilder builder) {
+    String proxyHost = null;
+    Integer proxyPort = null;
+    String proxyUser = null;
+    String proxyPassword = null;
+    if (config.isUseSystemProperties()) {
+      builder.useSystemProperties();
+      String protocol = System.getProperty("https.proxyHost") != null ? "https" : "http";
+      proxyHost = System.getProperty(protocol + ".proxyHost");
+      proxyPort = Integer.parseInt(System.getProperty(protocol + ".proxyPort"));
```

Comment:
If the client has set UseSystemProperties to true, they must ideally have these system properties set, otherwise these would fail with NPE when we try to parseInt from null/build an AuthScope with the relevant error i.e. xx is invalid/null